src/systemctl/systemctl.c

```diff
@@ -5668,6 +5713,12 @@ static int enable_unit(int argc, char *argv[], void *userdata) {
 
                 polkit_agent_open_if_enabled();
 
+                if (STR_IN_SET(verb, "mask", "unmask")) {
+                        r = unit_exists(*names, verb);
+                        if (r != 0)
+                                return r;
```

Comment:
On Wed, Jun 15, 2016 at 09:44:18AM -0700, Susant Sahani wrote:
> > @@ -5668,6 +5713,12 @@ static int enable_unit(int argc, char *argv[], void *userdata) {
> >  
> >                  polkit_agent_open_if_enabled();
> >  
> > +                if (STR_IN_SET(verb, "mask", "unmask")) {
> > +                        r = unit_exists(*names, verb);
> > +                        if (r != 0)
> > +                                return r;
> 
> @keszybz  Thanks . IMHO it looks to me if some one else going to install a package belongs to admin group may be for some reason we don't want to start it . I am still confused why would we install a package that we don't want to start and we don't know whether this is going to be in install or not . This is really is a bogus argument but admin can always rename it and start even if it's masked :)

For example because a package might be pulled in through dependencies.
So a package might be added without being "explicitly requested".